### PR TITLE
remove redundant StatusCode checks from many tests

### PR DESCRIPTION
**Pull request by S11001001**
* [x] ensure `parseResponse`'s StatusCode is truly redundant
* [x] remove it from the signature
* [x] fix everything downstream
* [x] refactor `resultContractId`

### CODE REVIEW

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTestFuns.scala [hunk 1]
```diff
@@ -284,25 +283,36 @@ trait AbstractHttpServiceIntegrationTestFuns
...
       private val self: Future[(A, JsValue)]
   ) {
-    def parseResponse[Result: JsonReader]: Future[(A, domain.SyncResponse[Result])] =
-      self.map(_ map (decode1[domain.SyncResponse, Result](_).fold(e => fail(e.shows), identity)))
+    def parseResponse[Result: JsonReader]: Future[domain.SyncResponse[Result]] =
+      self.map { case (status, jsv) =>
+        val r = decode1[domain.SyncResponse, Result](jsv).fold(e => fail(e.shows), identity)
+        r.status should ===(status)
+        r
+      }
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTestFuns.scala:291]: This is the focus of this PR. We thought we might have endpoints with different HTTP status and "status" field in the JSON response, so returned both here, but in practice (at least for all our tests) we don't do that. So we can test that it's really redundant every time here, and discard.

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 2]
```diff
@@ -441,11 +435,11 @@ abstract class AbstractHttpServiceIntegrationTestTokenIndependent
...
               .map(inside(_) {
                 case Seq(
-                      (StatusCodes.OK, jsVal1 @ domain.OkResponse(acl1 @ List(ac), _, _)),
-                      (StatusCodes.OK, jsVal2 @ domain.OkResponse(acl2, _, _)),
+                      jsVal1 @ domain.OkResponse(acl1 @ List(ac), _, StatusCodes.OK),
+                      jsVal2 @ domain.OkResponse(acl2, _, _),
                     ) =>
-                  jsVal1 shouldBe jsVal2
                   acl1 shouldBe acl2
+                  jsVal1 shouldBe jsVal2
```
Comment (S11001001) [inline ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:442]: This ordering makes more sense for test failure reporting.

File: ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTestFuns.scala (hunk 1 above)
Comment (ray-roestenburg-da) [reply ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTestFuns.scala:291]: nice cleanup!

**Review by ray-roestenburg-da — APPROVED** [review]
LGTM

**Review by chunlokling-da — APPROVED** [review]

**Pull request merged**